src/rdkafka_offset.c

```diff
@@ -795,15 +795,32 @@ void rd_kafka_offset_reset (rd_kafka_toppar_t *rktp, int64_t err_offset,
                    if available */
                 if (offset == RD_KAFKA_OFFSET_BEGINNING &&
                     rktp->rktp_lo_offset >= 0) {
+                        rd_kafka_dbg(rktp->rktp_rkt->rkt_rk, TOPIC, "OFFSET",
```

Comment:
haha, i preemptively didn't put debug statements in these if blocks because i sensed you'd have DRY complaints.

there is now no reason not to return early from these blocks so we should do that.

Reply:
Due to separation of concerns we can't assume that next_offset_handle() does not trigger an offset query, so I'll keep the fallthru now to be safe, but adjusting the duplicatory logs.